src/sentry/api/endpoints/project_rule_actions.py

```diff
@@ -60,7 +65,59 @@ def post(self, request: Request, project) -> Response:
             project, platform=project.platform, default="javascript", tagged=True
         )
 
+        if features.has(
+            "projects:verbose-test-alert-reporting", project=project, user=request.user
+        ):
+            return self.execute_future_on_test_event(test_event, rule)
+        else:
+            # Old existing behavior to wrap this in a handler which buries exceptions
+            for callback, futures in activate_downstream_actions(rule, test_event).values():
+                safe_execute(callback, test_event, futures)
+            return Response()
+
+    def execute_future_on_test_event(
+        self,
+        test_event: GroupEvent,
```

Comment:
This type is completely wrong, (`create_sample_event` return `Event | None`, neither of which are compatible with `activate_downstream_actions`. I'll see if I can update `activate_downstream_actions` to take a `BaseEvent` in the future, since I don't think it makes sense to make this change now.